### PR TITLE
hardcoded roles refatored to enum

### PR DESCRIPTION
**Pull request by RomanKlymus**

### CODE REVIEW

File: src/main/java/com/softserve/teachua/constants/RoleData.java [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 public enum RoleData {
     ADMIN("ADMIN"),
-    USER("USER");
+    USER("USER"),
+    MANAGER("MANAGER");
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/constants/RoleData.java:6]: Added role manager to enum, just like it is in DB

File: src/main/java/com/softserve/teachua/config/SecurityConfig.java [hunk 2]
```diff
@@ -26,27 +24,25 @@
...
-
 @Configuration
 @EnableWebSecurity
 @EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
 
-    @Autowired
+    private final static String ADMIN = RoleData.ADMIN.getRoleName();
+    private final static String USER = RoleData.USER.getRoleName();
+    private final static String MANAGER = RoleData.MANAGER.getRoleName();
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/config/SecurityConfig.java:34]: String variables for roles

File: src/main/java/com/softserve/teachua/config/SecurityConfig.java [hunk 3]
```diff
@@ -26,27 +24,25 @@
...
+    private final JwtFilter jwtFilter;
     private CustomUserDetailsService customUserDetailsService;
-
-    @Autowired
     private CustomOAuth2UserService customOAuth2UserService;
-
-    @Autowired
     private OAuth2AuthenticationSuccessHandler oAuth2AuthenticationSuccessHandler;
-
-    @Autowired
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/config/SecurityConfig.java:45]: move autowired annotation from variable to setter (watch below)

File: src/main/java/com/softserve/teachua/config/SecurityConfig.java [hunk 4]
```diff
@@ -210,4 +199,29 @@ protected void configure(HttpSecurity http) throws Exception {
                 .addFilterBefore(jwtFilter, UsernamePasswordAuthenticationFilter.class)
                 .logout().logoutRequestMatcher(new AntPathRequestMatcher("/signout")).logoutSuccessUrl("/signin");
     }
+
+    @Autowired
+    public void setCustomUserDetailsService(CustomUserDetailsService customUserDetailsService) {
+        this.customUserDetailsService = customUserDetailsService;
+    }
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/config/SecurityConfig.java:206]: and here are setters to autowire beans

**Review by VictoriaAmazonka — APPROVED** [review]

**Pull request merged**